yaks/spec/unit/yaks/format/json_api_spec.rb

```diff
@@ -38,4 +39,92 @@
     end
   end
 
+  context 'with a self link' do
+    let(:resource) {
+      Yaks::Resource.new(
+          type: 'wizard',
+          links: [
+              Yaks::Resource::Link.new(rel: :self, uri: '/the/self/link')
+          ]
+      )
+    }
+    it 'should use the self link in output' do
+      expect(format.call(resource)).to eql(
+         {data: [
+             {
+                 type: :wizards,
+                 href: '/the/self/link'
+             }
+         ]
+         }
+      )
+
+    end
+  end
+
+  context 'with subresources' do
+    let(:resource) {
+      Yaks::Resource.new(
+          type: 'wizard',
+          subresources: [
+              Yaks::Resource.new(type: 'spell', attributes: {id: 777, name: 'Lucky Sevens'})
+          ]
+      )
+    }
+    it 'should include links and linked' do
+      expect(format.call(resource)).to eql(
+         {
+           data: [
+             {
+               type: :wizards,
+               links: {'spell'  => {type: 'spells', id: 777}}
+             }
+           ],
+           linked: [{type: :spells, id: 777, name: 'Lucky Sevens'}]
+         }
+      )
+    end
+
+  end
+
+  context 'with null subresources' do
+    let(:resource) {
+      Yaks::Resource.new(
+          type: 'wizard',
+          subresources: [
+              Yaks::NullResource.new
+          ]
+      )
+    }
+    it 'should not include links' do
+      expect(format.call(resource)).to eql(
+         {data: [
+             {
+                 type: :wizards,
+             }
+         ]
+         }
+      )
```

Comment:
minor remark. I probably would collapse this all to one line

``` ruby
expect(format.call(resource)).to eql( {data: [{type: :wizards}]} )
```

or even

``` ruby
expect(format.call(resource)).to eql data: [{type: :wizards}]
```

If the result still comes in in under 80 chars per line that's more than good enough.